### DISCUSSION TITLE
Ignore any exception from Http facade

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#43236] [Ideas]
```php
Http::timeout(1)->get('https://httpstat.us/200?sleep=5000')
```

I use Http facade in many places in my app, Is there possible to ignore any exception when using it without try catch ? note that it throw exception on ConnectException:

https://github.com/laravel/framework/blob/76b407eafb16dfd4a553f3c437fbbc6d8268e3a1/src/Illuminate/Http/Client/PendingRequest.php#L809

**Comment by foremtehan** [comment]
I dont think it's possible ,nvm

**Comment by bretto36** [comment]
You have to manually call the throw() method on the Http facade don't you?

Http::get('https://example.com')->throw()->object();

Could you not put in the throw method call?

**Comment by foremtehan** [comment]
@bretto36 run this `Http::timeout(1)->get('https://httpstat.us/200?sleep=5000')` the ConnectionException thrown no matter what.

**Comment by foremtehan** [comment]
using 20 rescure with Http ? No.

**Comment by hugo-lovighi** [comment]
To ignore Http exception I use this:
`Http::throwIf(false)->get('https://.....')`

**Reply by foremtehan** [reply]
Try this and you will get a throw:
`\Http::throwIf(false)->timeout(1)->get('https://httpstat.us/200?sleep=5000');`

**Comment by DvDty** [comment]
Might be happening because of how guzzle exception hierarchy is: https://docs.guzzlephp.org/en/stable/quickstart.html#exceptions

Timeout exceptions are not under request.

On the question: what about adding a macro method to the Http facade, that makes the request call but is surrounded by a try catch?
